File: euler-problem12/euler.js
// following line can be useful to test the script inside a browser
typeof module === 'undefined' && console.time('main');

function main() {

    function factorCount (n)
    {
        var square = Math.sqrt(n);
        var isquare = ~~square;
        var count = isquare == square ? -1 : 0;
        var candidate;
        for (candidate = 1; candidate <= isquare; candidate ++)
            if (0 == n % candidate) count += 2;
        
        return count;
    }


    triangle = 1;
    index = 1;
    while (factorCount (triangle) < 1001)
    {
        index ++;
        triangle += index;
    }
    console.log(triangle);

}

main();

typeof module === 'undefined' && console.timeEnd('main');

File: euler-problem12/euler.java

public class euler {

    public static int factorCount(long n) {
        double square = Math.sqrt(n);
        int isquare = (int) square;
        int count = isquare == square ? -1 : 0;
        long candidate;
        for (candidate = 1; candidate <= isquare; candidate++) {
            if (0 == n % candidate) {
                count += 2;
            }
        }
        return count;
    }

    public static void main(String args[]) {
        long triangle = 1;
        int index = 1;
        while (euler.factorCount(triangle) < 1001) {
            index++;
            triangle += index;
        }

        System.out.println(triangle);
    }
}
